https.go

```diff
@@ -494,7 +494,7 @@ func (proxy *ProxyHttpServer) connectDialProxyWithContext(ctx *ProxyCtx, proxyHo
 		return nil, err
 	}
 
-	c, err := proxy.connectDialContext(ctx, "tcp", proxyHost)
+	c, err := proxy.connectDialContext(ctx, "tcp", proxyURL.Host)
```

Comment:
I've read through this a couple of times, and think that it would be more straightforward to just remove the call to `url.Parse` continue to pass `proxyHost` directly.

Reply:
@cds2-stripe: yup, you're probably right here. That does make things simpler. 

It means I'll need to remove this block though:
```
if proxyURL.Scheme == "https" {	
		c = tls.Client(c, proxy.Tr.TLSClientConfig)	
	}
```

In the previous implementation, I'm fairly certain that was unreachable code though? `httpsProxyFromEnv` will never return a url with a scheme.

Reply:
Oh wow, nice catch! I completely glossed over this.

You're correct. The proxy string returned from `httpsProxyFromEnv` never included a scheme, and `url.Parse` does not infer one even if we're using the default HTTPS port. This means that we'd fall back to the default `tls.Client` which is a bug.

What do you think about keeping the `url.Parse` and changing the return statement so that it includes the scheme?  This would mean that for connections to HTTPS proxies we would hit the code path that sets the correct TLS client config. Something like:

```
	return fmt.Sprintf("%s//%s:%s", proxyURL.Scheme, proxyURL.Hostname(), service), nil
```

Reply:
Makes sense to me! Updated the PR with this suggestion.